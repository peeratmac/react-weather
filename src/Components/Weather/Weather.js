import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setWeatherInfo } from '../../actions';
import { fetchUsingStationID } from '../../apiCalls';

export class Weather extends Component {
  grabWeatherDataWithStationID = () => {
    const weather = fetchUsingStationID(this.props.currentStation);
    console.log(weather);
    this.props.setWeatherInfo(weather);
  };

  render() {
    console.log(this.props);
    return (
      <section>
        <p>Weather</p>
        <h2>
          {/* {this.props.weatherInfo.map(
            x => x.consolidated_weather[0].weather_state_name
          )} */}
        </h2>
        <p>Weather</p>
      </section>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  setWeatherInfo: weatherInfo => dispatch(setWeatherInfo(weatherInfo))
});

const mapStateToProps = state => ({
  cities: state.cities,
  stationIDs: state.stationIDs,
  currentStation: state.currentStation,
  weatherInfo: state.weatherInfo
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Weather);
