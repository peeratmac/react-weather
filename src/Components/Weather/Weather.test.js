import React from 'react';
import { shallow } from 'enzyme';
import { Weather } from './Weather';

describe('WEATHER', () => {
  let wrapper;

  const mockSetWeatherInfo = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Weather setWeatherInfo={mockSetWeatherInfo} />);
  });

  it('should match the Nav Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
