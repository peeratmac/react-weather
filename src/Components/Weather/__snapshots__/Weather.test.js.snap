// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WEATHER should match the Nav Snapshot 1`] = `
<section>
  <p>
    Weather
  </p>
  <h2 />
  <p>
    Weather
  </p>
</section>
`;
